### PR TITLE
cleanup(pubsub): set attribute only when present

### PR DESCRIPTION
**Pull request by alevenberg**
Closes #[12620](https://github.com/googleapis/google-cloud-cpp/issues/12620)

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12636)

### CODE REVIEW

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/internal/publisher_tracing_connection_test.cc [hunk 1]
```diff
@@ -135,6 +136,35 @@ TEST(PublisherTracingConnectionTest, PublishSpanOnError) {
...
+                      .get();
+
+  EXPECT_STATUS_OK(response);
+  EXPECT_THAT(span_catcher->GetSpans(),
+              ElementsAre(AllOf(
+                  SpanHasInstrumentationScope(), SpanKindIsProducer(),
+                  SpanNamed("projects/test-project/topics/test-topic send"),
+                  SpanWithStatus(opentelemetry::trace::StatusCode::kOk),
+                  Not(SpanHasAttributes(OTelAttribute<std::string>(
+                      "messaging.pubsub.ordering_key", ""))))));
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/publisher_tracing_connection_test.cc:165]: To be more general, I would `s/""/_/`

(and maybe you need a `using ::testing::_;` somewhere.

Comment (alevenberg) [reply google/cloud/pubsub/internal/publisher_tracing_connection_test.cc]: Done

File: google/cloud/pubsub/internal/publisher_tracing_connection.cc [hunk 2]
```diff
@@ -44,18 +44,20 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> StartPublishSpan(
...
+                                  {sc::kMessagingSystem, "pubsub"},
+                                  {sc::kMessagingDestinationName, topic},
+                                  {sc::kMessagingDestinationTemplate, "topic"},
+                                  {"messaging.message.total_size_bytes",
+                                   static_cast<std::int64_t>(MessageSize(m))},
+                              },
+                              options);
+
+  if (!m.ordering_key().empty()) {
+    span->SetAttribute("messaging.pubsub.ordering_key", m.ordering_key());
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/publisher_tracing_connection.cc:59]: FYI: If we set the attribute after the span is created, our customers can no longer use the ordering_key attribute to make sampling decisions. Is that ok?

Comment (alevenberg) [reply google/cloud/pubsub/internal/publisher_tracing_connection.cc:59]: Ack. I think it should be ok, but it's a good call out.

**Pull request merged**